frontend/cypress/support/commands.ts

```diff
@@ -41,8 +41,212 @@
-// eslint-disable-next-line no-restricted-imports
-import * as cypressImageDiffCommand from 'cypress-image-diff-js/dist/command';
 
-cypressImageDiffCommand();
+import type {
+    JsonObject,
+    UUID,
+    SocketResponse,
+    ExerciseAction,
+} from 'digital-fuesim-manv-shared';
+import { io } from 'socket.io-client';
+
+export function dragToMap(
+    elementSelector: string,
+    offset?: { x: number; y: number }
+) {
+    const mapSelector = '[data-cy=openLayersContainer]';
+
+    cy.get(elementSelector).first().trigger('mousedown');
+
+    return (
+        cy
+            .get(mapSelector)
+            /**
+             * We have to force the click here, as the img
+             * of the element being dragged is in front of the map
+             **/
+            .trigger('mousemove', { force: true, ...offset })
+            .click({ force: true })
+    );
+}
+
+export function store() {
+    return cy.window().its('cypressTestingValues', { log: false }).its('store');
+}
+
+export function getState() {
+    return cy.wrap(
+        new Promise((resolve) => {
+            cy.store()
+                .invoke({ log: false }, 'select', 'application')
+                .invoke({ log: false }, 'subscribe', (state: any) =>
+                    resolve(state)
+                )
+                .log('get state');
+        }),
+        { log: false }
+    );
+}
+
+export function createExercise() {
+    cy.visit('/');
+    cy.window()
+        .its('cypressTestingValues', { log: false })
+        .its('backendBaseUrl')
+        .as('backendBaseUrl');
+    cy.window()
+        .its('cypressTestingValues', { log: false })
+        .its('websocketBaseUrl')
+        .as('websocketBaseUrl');
+
+    cy.get('@backendBaseUrl', { log: false }).then((backendBaseUrl) =>
+        cy
+            .request('POST', `${backendBaseUrl}/api/exercise`)
+            .its('body')
+            .as('createBody')
+    );
+    cy.get('@createBody', { log: false }).its('trainerId').as('trainerId');
+    cy.get('@createBody', { log: false })
+        .its('participantId')
+        .as('participantId');
+    return cy;
+}
+
+export function joinExerciseAsTrainer() {
+    cy.get('@trainerId', { log: false }).then((trainerId) =>
+        cy.visit(`exercises/${trainerId}`)
+    );
+    cy.get('[data-cy=joinExerciseModalButton]').click();
+    return cy;
+}
+
+export function joinExerciseAsParticipant() {
+    cy.get('@participantId', { log: false }).then((participantId) =>
+        cy.visit(`exercises/${participantId}`)
+    );
+    cy.get('[data-cy=joinExerciseModalButton]').click();
+    return cy;
+}
+
+export function initializeParticipantSocket() {
+    cy.get('@websocketBaseUrl', { log: false }).then(
+        (websocketBaseUrl: any) => {
+            cy.wrap(io(websocketBaseUrl, { transports: ['websocket'] })).as(
+                'participantSocket'
+            );
+        }
+    );
+
+    cy.get('@participantId', { log: false }).then((participantId: any) => {
```

Comment:
Do you need to use any here? same for Trainer

Reply:
Yup, I actually need to, as cypress thinks that I'm storing Jquery elements and there is no real cast to these.